### PR TITLE
chore(other): adds commitlint and commitzen

### PR DESCRIPTION
**Pull request by waldenraines**
## PR Type

* [ ] Bug fix
* [ ] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactor (i.e., no functional changes)
* [ ] Build related changes
* [x] CI related changes
* [ ] Documentation changes
* [ ] Other (please describe below):

## Description

Add commitlint and commitzen to the project in preparation for semantic releases.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

### CODE REVIEW

**Comment by waldenraines** [comment]
Peek 2020-04-22 11-57

**Comment by bucko13** [comment]
Cool!

**Comment by waldenraines** [comment]
A huge added benefit of this style of commit messages that I forgot to mention is that it allows the automatic generation of changelog comments.

File: .cz-config.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const scopes = [
+  "card",
+  "clientpicker",
+  "createaddress",
+  "hermit",
+  "slices",
+  "testrunner",
+  "wallet",
+  "other",
```
Comment (waldenraines) [inline .cz-config.js:38]: Please have a look at these and see what I need to add/remove.

File: .cz-config.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  confirmCommit: `Are you sure you want to proceed with the commit above?`
+};
+
+const scopes = [
+  "card",
+  "clientpicker",
+  "createaddress",
+  "hermit",
+  "slices",
```
Comment (destrys) [inline .cz-config.js:35]: what are slices?

Comment (waldenraines) [reply .cz-config.js:35]: I think they're the little table panel things that @bucko13 put in.

Comment (bucko13) [reply .cz-config.js:35]: a "slice" is the name we're using for a cross section of braid at a particular path

**Pull request merged**